pymc3/distributions/multivariate.py

```diff
@@ -1911,39 +1867,26 @@ def _quaddist(self, value):
         else:
             onedim = False
 
-        delta = value - self.mu
-        if self._isEVD:
-            sqrt_quad = kron_dot(self.QTs, delta.T)
-            sqrt_quad = sqrt_quad / at.sqrt(self.eigs[:, None])
-            logdet = at.sum(at.log(self.eigs))
-        else:
-            sqrt_quad = kron_solve_lower(self.chols, delta.T)
-            logdet = 0
-            for chol_size, chol_diag in zip(self.sizes, self.chol_diags):
-                logchol = at.log(chol_diag) * self.N / chol_size
-                logdet += at.sum(2 * logchol)
+        delta = value - mu
+
+        chols = list(map(cholesky, covs))
+        chols = [_chol.eval() for _chol in chols]
+        sizes = at.as_tensor_variable([chol.shape[0] for chol in chols])
+        N = at.prod(sizes)
+        chol_diags = list(map(at.diag, chols))
+        sqrt_quad = kron_solve_lower(chols, delta.T)
+        logdet = 0
+        for chol_size, chol_diag in zip(sizes, chol_diags):
+            logchol = at.log(chol_diag) * N / chol_size
+            logdet += at.sum(2 * logchol)
```

Comment:
So for now I removed the Eigen vector decomposition from here, and this works for all three cases of `cov`, `chol` and `evd`. However this leaves us unable to use `sigma` i.e. Gaussian white noise, over here in `logp`. 

Regardless can someone verify if this implementation is mathematically correct for the `evd` case ? (The test pass, however, I'm not too sure if this approach is mathematically correct for that case.)